src/components/GlobalModal.js

```diff
@@ -82,6 +97,12 @@ export const GlobalModal = ({ children }) => {
     // Same problem happens with the class jquery adds to the body,
     // causing the app to stop scrolling. We can just remove it
     $('body').removeClass('modal-open');
+
+    // Managing the modal lifecycle, if the string parameter is offered
```

Comment:
This was the best solution I found to managing the DOM of the new modals, since they're using the functional components pattern and not using classes and thus unable to implement the `componentWillUnmount` in a similar way to what's already available on the codebase.

With other solutions within the same component I was constantly having bugs managing the lifecycle of the modals and thought it would be a good idea to centralize this DOM management on the modals parent instead of inside each of them.

This way it's not even necessary to import jQuery's `$` function into the modals anymore, abstracting React code away from the raw DOM manipulation.

Reply:
To implement the exact same behavior as `componentWillUnmount` on hooks you can:

```js
useEffect(() => {
  return () => {
    // componenWillUnmount logic here
  };
}, []);
```

The second argument (an empty array) ensures that the returned function only runs when the component will unmount.
This change to use a global modal was made to fix a couple of issues with modal implementations on the wallet, I think we should understand which issues were happenning and if these changes can "revive" those issues.

From what I remember, one specific issue was with the modal lifecycle, that sometimes the modal would close due to a change of screens but the backdrop would not go away, and the backdrop would prevent any interactions with the wallet. The user would then need to close the wallet and open again.

Reply:
I've made some tests by programatically generating failures while with the modal open, and even reloading the wallet itself simulating a network change. In all these situations the user was able to just dismiss the modal with a click or using the `Esc` key, as the video below shows:

https://user-images.githubusercontent.com/1299409/217548718-91458d15-50dd-4ed5-979b-f0586e6d8105.mp4